modules/command-engine/core/src/main/scala/surge/core/commondsl/AbstractSurgeEventBusinessLogic.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>
+
+package surge.core.commondsl
+
+import com.typesafe.config.Config
+import io.opentracing.Tracer
+import io.opentracing.noop.NoopTracerFactory
+import surge.core.event.{ AggregateEventModelCoreTrait, SurgeEventKafkaConfig }
+import surge.core.{ SurgeAggregateFormatting, SurgeEventWriteFormatting }
+import surge.kafka.KafkaTopic
+import surge.metrics.Metrics
+
+abstract class AbstractSurgeEventBusinessLogic[AggId, Agg, Event](private val config: Config) {
+
+  def aggregateName: String
+
+  def stateTopic: KafkaTopic
+
+  def aggregateFormatting: SurgeAggregateFormatting[Agg]
+
+  def eventWriteFormatting: SurgeEventWriteFormatting[Event]
```

Comment:
Another odd difference, we're using the `aggregateFormatting`/`eventWriteFormatting` here and the `writeFormatting`/`aggregateReadFormatting` there.  Maybe we want to be more explicit - `aggregateReadFormatting`, `aggregateWriteFormatting`, and `eventWriteFormatting`? That way you could override with any of the combination formatters.